api/controllers/projectV2.js

```diff
@@ -0,0 +1,580 @@
+
+            let project = await projectDAO.getProject(SECURE_ROLES, projectId);
+
+            if(project)
+            {
+                project = await projectDAO.updateExtension(args.swagger.params.auth_payload.preferred_username, extension, project);
+                project = projectDAO.projectHateoas(project, roles);
+                return Actions.sendResponseV2(res, 200, project);
+            }
+            else
+            {
+                return Actions.sendResponseV2(res, 404, { status: 404, message: 'Project ' + projectId + ' not found.'});
+            }
+        }
+        else
+        {
+            throw Error('Invalid request');
+        }
+    }
+    catch (e)
+    {
+        defaultLog.error('### Error in {PUT}/Projects/{id}/Extensions :', e);
+        return Actions.sendResponseV2(res, 500, { code: '500', message: 'Internal Server Error', self: 'Api/Projects{id}/Extensions' });        
+    }
+    finally
+    {
+        defaultLog.debug('<<< {PUT}/Projects/{id}/Extensions');
+    }
+};
+
+// DELETE (Protected Only, deleteExtension)
+exports.deleteProjectExtension = async function (args, res, next)
+{
+    defaultLog.debug('>>> {DELETE}/Projects/{id}/Extensions');
+
+    try
+    {
+        if (args.swagger.params.hasOwnProperty('projId') && args.swagger.params.hasOwnProperty('extension'))
+        {
+            let projectId = args.swagger.params.projId.value;
+            let extension = JSON.parse(args.swagger.params.item.value);
+
+            defaultLog.debug(' Updating extension on project ' + projectId);
+            defaultLog.debug(' Extension: ' + JSON.stringify(extension));
+
+            let project = await projectDAO.getProject(SECURE_ROLES, projectId);
+
+            if(project)
+            {
+                project = await projectDAO.updateExtension(args.swagger.params.auth_payload.preferred_username, extension, project);
+                project = projectDAO.projectHateoas(project, roles);
+                return Actions.sendResponseV2(res, 200, project);
+            }
+            else
+            {
+                return Actions.sendResponseV2(res, 404, { status: 404, message: 'Project ' + projectId + ' not found.'});
+            }
+        }
+        else
+        {
+            throw Error('Invalid request');
+        }
+    }
+    catch (e)
+    {
+        defaultLog.error('### Error in {DELETE}/Projects/{id}/Extensions :', e);
+        return Actions.sendResponseV2(res, 500, { code: '500', message: 'Internal Server Error', self: 'Api/Projects{id}/Extensions' });
+    }
+    finally
+    {
+        defaultLog.debug('<<< {DELETE}/Projects/{id}/Extensions');
+    }
+};
+
+exports.fetchFeaturedDocuments = async function (args, res, next) 
+{
+    defaultLog.debug('>>> {GET}/Public/Projects/{id}/FeaturedDocuments');
+
+    try
+    {
+        if (args.swagger.params.projId && args.swagger.params.projId.value) 
+        {
+            let projectModel = mongoose.model('Project');
```

Comment:
Should this be using ` projectDAO.getProject(SECURE_ROLES, projectId);`?

Reply:
Yep. I copied these from the recent addition for featured docs but haven't optomized them for the v2 endpoints. These endpoints will be adjusted to fit with the new ones